src/main/java/io/boomerang/security/util/MultiReadHttpServletRequest.java

```diff
@@ -28,18 +27,16 @@
 import javax.servlet.http.Part;
 import org.springframework.util.StreamUtils;
 
-/*
- * Reference: https://www.baeldung.com/spring-reading-httpservletrequest-multiple-times
- */
 public class MultiReadHttpServletRequest implements HttpServletRequest {
 
   private final byte[] buffer;
   private final HttpServletRequest request;
 
   public MultiReadHttpServletRequest(HttpServletRequest request) throws IOException {
     this.request = request;
-    InputStream requestInputStream = request.getInputStream();
-    this.buffer = StreamUtils.copyToByteArray(requestInputStream);
+    ByteArrayOutputStream bufferStream = new ByteArrayOutputStream();
+    StreamUtils.copy(request.getInputStream(), bufferStream);
+    this.buffer = bufferStream.toByteArray();
```

Comment:
Can you run me through this change?

Reply:
I think this is done by @corcoja . As far as I understand it, there was a wrapper Type 'unboxed': InputStream. This is in relation with `getReader` method.

Reply:
Same comment as below 😅 This PR changes the code snippet below. Compared to the old code, it looks like a longer way of copying one array to another, but @AndreiOla should know better.

```java
    InputStream requestInputStream = request.getInputStream();
    this.buffer = StreamUtils.copyToByteArray(requestInputStream);
```

Reply:
Well, it seems I messed some things up, when I extracted the changes. I reverted them on how they were done by @tlawrie .